### PR TITLE
Allows users to submit rfc822 formatted email addresses

### LINKED ISSUE #277: Inconsistent format of email from header “name <email>”
#### Issue Summary

Current implementation restrict use default `email_from` behavior. For more info please see https://tools.ietf.org/html/rfc2822#section-3.4

#### Steps to Reproduce
1. When user try to create `Email` instance from https://github.com/sendgrid/sendgrid-python/blob/master/sendgrid/helpers/mail/mail.py#L173-L179. You must specify 
`from_email` and `from_name` as seperate parametrs
```
    def get(self):
        email = {}
        if self.name != None:
            email["name"] = self.name
        if self.email != None:
            email["email"] = self.email
        return email

from_email = Email("test@example.com", "Example Name")
```
As a result client should split `from_email`:
```
try:
    import rfc822
except ImportError:
    import email.utils as rfc822


from_name, from_email = rfc822.parseaddr(email.from_email)
# Python sendgrid client should improve
# sendgrid/helpers/mail/mail.py:164
if not from_name:
    from_name = None
Mail(from_email, from_name)
```

Proposal to how to improve this bahaviour.
Because if user now create `Email` from:
```
from_email = Email("Andrii Soldatenko <hello@asoldatenko.com>")
```
**Actual result:**
image

#### Technical details:

* sendgrid-python Version: master
* Python Version: all

#### References:
- https://github.com/sendgrid/sendgrid-python/issues/277
- RFC: https://tools.ietf.org/html/rfc2822#section-3.4
- Django docs: https://docs.djangoproject.com/en/1.10/topics/email/#emailmessage-objects

**NOTE:**
In `sengrid-python==1.4` this feature is working

### PR DESCRIPTION
**Pull request by mbernier**
Closes #277

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-python?pullRequest=348) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-python?pullRequest=348) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-python?pullRequest=348) it.</sub>

**Comment by mbernier** [comment]
@thinkingserious I am not sure what happened with the Travis build... looks like there's an env problem?

File: sendgrid/helpers/mail/mail.py [hunk 1]
```diff
@@ -260,13 +260,15 @@ def add_custom_arg(self, custom_arg):
...
 
     def __init__(self, email=None, name=None):
-        self._name = None
-        self._email = None
-
-        if email is not None:
-            self.email = email
-        if name is not None:
-            self.name = name
+        if not name
```
Comment (thinkingserious) [inline sendgrid/helpers/mail/mail.py]: Should be `if not name:`

**Comment by mbernier** [comment]
Updated, thanks!

**Comment by mbernier** [comment]
@thinkingserious I added a ridiculous number of tests and even found something I missed. You win. Fine. Are you happy!!?

**Comment by thinkingserious** [comment]
Hello @mbernier,<br /><br/>Thanks again for the PR!<br />                                              <br />It's <a href="https://sendgrid.com/blog/announcing-hacktoberfest-2017">HACKTOBERFEST</a>! We want to show our appreciation by sending                                              you some special Hacktoberfest swag. If you have not already, could you please fill out                                              <a href="https://dx.sendgrid.com/swag">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by mbernier** [comment]
It's MERGED!!!

**Pull request merged**